src/Workspaces/Core/Desktop/Workspace/MSBuild/ProjectFile/ProjectFile.cs

```diff
@@ -192,6 +192,8 @@ private static Task<MSB.Execution.BuildResult> BuildAsync(MSB.Execution.BuildMan
                         taskSource.TrySetCanceled();
                     }
                 });
+
+                taskSource.Task.SafeContinueWith(_ => registration.Dispose(), CancellationToken.None, TaskContinuationOptions.ExecuteSynchronously, TaskScheduler.Default);
```

Comment:
> Just to verify, but registration is safe to double dispose?

It should be, but either way it seems there's no longer any need for the second call so I can remove it.

> i'm not a huge fan of hte SafeContinueWith approach.

It reduces the overhead of the cleanup by allowing the use of `ExecuteSynchronously`.